app/views/topology/service.js

```diff
@@ -1191,9 +1191,12 @@ YUI.add('juju-topology-service', function(Y) {
       // annotations.
       var new_service_boxes = Y.Object.values(topo.service_boxes)
       .filter(function(boundingBox) {
-            var annotations = boundingBox.model.get('annotations');
-            return ((!Y.Lang.isNumber(boundingBox.x) &&
-                !(annotations && annotations['gui-x'])));
+            if (boundingBox.model) {
+              var annotations = boundingBox.model.get('annotations');
+              return ((!Y.Lang.isNumber(boundingBox.x) &&
+                  !(annotations && annotations['gui-x'])));
+            }
+            return false;
```

Comment:
This was a driveby, meant to put a GH comment on it but spaced it.  Gets rid of one of the three regularly occurring errors with the ECS (the other two you describe below)